runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -6937,16 +6937,57 @@ J9::Z::TreeEvaluator::VMifInstanceOfEvaluator(TR::Node * node, TR::CodeGenerator
    return NULL;
    }
 
+TR::Register *
+J9::Z::TreeEvaluator::generateCheckForValueTypeMonitorEnterOrExit(TR::Node *node, TR::Register *objReg, TR::LabelSymbol* mergeLabel, TR::LabelSymbol *helperCallLabel, TR::CodeGenerator *cg)
+   {
+   TR::Register *objectClassReg = cg->allocateRegister();
+   TR::TreeEvaluator::genLoadForObjectHeadersMasked(cg, node, objectClassReg, generateS390MemoryReference(objReg, TR::Compiler->om.offsetOfObjectVftField(), cg), NULL);
+   TR::Register *classFlagsRegister = cg->allocateRegister();
```

Comment:
This register leaks. We never seem to call `cg->stopUsingRegister` on it.

Reply:
Yeah, I forgot to push latest changes, I found that while testing. https://github.com/eclipse/openj9/compare/c99f445c36fe76ead49f8c155b36812bf8df525b..097726868780ab77a37f3f823e2ba1d82900e0db contains the change.